File: examples/dart_example/lib/counter_example.dart

import 'package:flutter/material.dart';
import 'package:flutter_unreal/flutter_unreal.dart';

class CounterExample extends StatefulWidget {
  const CounterExample({super.key});

  @override
  State<CounterExample> createState() => _CounterExampleState();
}

class _CounterExampleState extends State<CounterExample> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  void _decrementCounter() {
    setState(() {
      _counter--;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.transparent,
      appBar: AppBar(backgroundColor: Colors.transparent),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const BlockPointer(
              child: Text(
                'You have pushed the button this many times:',
              )
            ),
            BlockPointer(child: Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            )),
          ],
        ),
      ),
      floatingActionButton: BlockPointer(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children:[
            FloatingActionButton(
              onPressed: _incrementCounter,
              tooltip: 'Increment',
              child: const Icon(Icons.add),
            ),
            const SizedBox(height: 10),
            FloatingActionButton(
              onPressed: _decrementCounter,
              tooltip: 'Decrement',
              child: const Icon(Icons.remove),
            )
          ]
        )
      )
    );
  }
}

File: examples/typescript_example/src/counter_example.ts

namespace flutter {
    export function CounterExample(): Widget {
        let _counter = 0
        return Builder({
            builder(context: BuildContext) {
                return Scaffold({
                    backgroundColor: Colors.transparent,
                    appBar: AppBar({backgroundColor: Colors.transparent}),
                    body: Center({
                        child: Column({
                            mainAxisAlignment: MainAxisAlignment.center,
                            children: [
                                BlockPointer({
                                    child: Text(
                                        'You have pushed the button this many times:',
                                    )
                                }),
                                BlockPointer({
                                    child: Text(
                                        `${_counter}`,
                                        {style: Theme.of(context).textTheme.headlineMedium},
                                    )
                                }),
                            ],
                        }),
                    }),
                    floatingActionButton: BlockPointer({
                        child: Column({
                            mainAxisSize: MainAxisSize.min,
                            children:[
                            FloatingActionButton({
                                onPressed() {
                                    _counter++
                                    markNeedsBuild(context)
                                },
                                tooltip: 'Increment',
                                child: Icon(Icons.add),
                            }),
                            SizedBox({height: 10}),
                            FloatingActionButton({
                                onPressed() {
                                    _counter--
                                    markNeedsBuild(context)
                                },
                                tooltip: 'Decrement',
                                child: Icon(Icons.remove),
                                })
                            ]
                        })
                    })
                })
            }
        })
    }
}
